messaging/MessagingExample/AppDelegate.m

```diff
@@ -187,6 +187,13 @@ - (void)tokenRefreshNotification:(NSNotification *)notification {
 
 // [START connect_to_fcm]
 - (void)connectToFcm {
+  if (![[FIRInstanceID instanceID] token]) {
+    return;
```

Comment:
Could you add a log message here that states "Won't connect since there is no token"?